src/main/java/org/reaktivity/nukleus/socks/internal/SocksController.java

```diff
@@ -0,0 +1,247 @@
+
+    public CompletableFuture<Long> routeClient(
+        String source,
+        long sourceRef,
+        String target,
+        long targetRef,
+        String dstAddrPort
+        )
+    {
+        long correlationId = controllerSpi.nextCorrelationId();
+        RouteFW route = routeRW.wrap(writeBuffer, 0, writeBuffer.capacity())
+            .correlationId(correlationId)
+            .role(b -> b.set(Role.CLIENT))
+            .source(source)
+            .sourceRef(sourceRef)
+            .target(target)
+            .targetRef(targetRef)
+            .extension(b -> b.set(visitRouteEx(dstAddrPort)))
+            .build();
+        return controllerSpi.doRoute(
+            route.typeId(),
+            route.buffer(),
+            route.offset(),
+            route.sizeof()
+        );
+    }
+
+    public CompletableFuture<Void> unrouteServer(
+        String source,
+        long sourceRef,
+        String target,
+        long targetRef,
+        String dstAddrPort
+        )
+    {
+        long correlationId = controllerSpi.nextCorrelationId();
+        UnrouteFW unroute = unrouteRW.wrap(writeBuffer, 0, writeBuffer.capacity())
+            .correlationId(correlationId)
+            .role(b -> b.set(Role.SERVER))
+            .source(source)
+            .sourceRef(sourceRef)
+            .target(target)
+            .targetRef(targetRef)
+            .extension(b -> b.set(visitRouteEx(dstAddrPort)))
+            .build();
+        return controllerSpi.doUnroute(
+            unroute.typeId(),
+            unroute.buffer(),
+            unroute.offset(),
+            unroute.sizeof()
+        );
+    }
+
+    public CompletableFuture<Void> unrouteClient(
+        String source,
+        long sourceRef,
+        String target,
+        long targetRef,
+        String dstAddrPort)
+    {
+        long correlationId = controllerSpi.nextCorrelationId();
+        UnrouteFW unroute = unrouteRW.wrap(writeBuffer, 0, writeBuffer.capacity())
+            .correlationId(correlationId)
+            .role(b -> b.set(Role.CLIENT))
+            .source(source)
+            .sourceRef(sourceRef)
+            .target(target)
+            .targetRef(targetRef)
+            .extension(b -> b.set(visitRouteEx(dstAddrPort)))
+            .build();
+        return controllerSpi.doUnroute(
+            unroute.typeId(),
+            unroute.buffer(),
+            unroute.offset(),
+            unroute.sizeof());
+    }
+
+    private Flyweight.Builder.Visitor visitRouteEx(
+        String dstAddrPort)
+    {
+        int lastColonIndex = dstAddrPort.lastIndexOf(":");
+        if (lastColonIndex == -1 || lastColonIndex == dstAddrPort.length() - 1)
+        {
+            throw new IllegalArgumentException("Missing colon from destination port and address, " + dstAddrPort);
+        }
+        String t0 = dstAddrPort.substring(0, lastColonIndex);
+        String t1 = dstAddrPort.substring(lastColonIndex + 1);
+        // TODO Ipv6 syntax should be supported
```

Comment:
Would it be possible to just call java.net.InetAddress.getByName(String) instead? That should take care of all cases I think and you could remove all this parsing code.